pkg/resource/deploy/source_eval.go

```diff
@@ -772,23 +772,40 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 
 	label := fmt.Sprintf("ResourceMonitor.RegisterResource(%s,%s)", t, name)
 
-	var providerRef providers.Reference
-	if custom && !providers.IsProviderType(t) || remote {
-		provider := req.GetProvider()
+	getProviderRef := func(provider string) (providers.Reference, error) {
```

Comment:
I'll probably extract this into a first-class function.

Reply:
Apparently this function already existed, so updated the code to use that instead.